bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```
@@ -0,0 +1,492 @@
+            logger.debug("Current unique address is: {}", properties.get(PROPERTY_UNIQUE_ADDRESS));
+            return;
+        }
+
+        ModbusEndpointThingHandler handler = getEndpointThingHandler();
+        if (handler == null) {
+            return;
+        }
+        getThing().setProperty(PROPERTY_UNIQUE_ADDRESS, handler.getUID().getAsString() + ":" + block.address);
+    }
+
+    /**
+     * Dispose the binding correctly
+     */
+    @Override
+    public void dispose() {
+        tearDown();
+    }
+
+    /**
+     * Unregister the poll task and release the endpoint reference
+     */
+    private void tearDown() {
+        unregisterPollTask();
+        unregisterEndpoint();
+    }
+
+    /**
+     * Returns the current slave id from the bridge
+     */
+    public int getSlaveId() {
+        return slaveId;
+    }
+
+    /**
+     * Get the endpoint handler from the bridge this handler is connected to
+     * Checks that we're connected to the right type of bridge
+     *
+     * @return the endpoint handler or null if the bridge does not exist
+     */
+    private @Nullable ModbusEndpointThingHandler getEndpointThingHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            logger.debug("Bridge is null");
+            return null;
+        }
+        if (bridge.getStatus() != ThingStatus.ONLINE) {
+            logger.debug("Bridge is not online");
+            return null;
+        }
+
+        ThingHandler handler = bridge.getHandler();
+        if (handler == null) {
+            logger.debug("Bridge handler is null");
+            return null;
+        }
+
+        if (handler instanceof ModbusEndpointThingHandler) {
+            ModbusEndpointThingHandler slaveEndpoint = (ModbusEndpointThingHandler) handler;
+            return slaveEndpoint;
+        } else {
+            logger.debug("Unexpected bridge handler: {}", handler);
+            throw new IllegalStateException();
```

Why would this throw an exception rather than return null?

---

Thanks you are now. Changed it for a more coherent behavior.